src/config/mod.rs

```diff
@@ -1986,6 +1989,28 @@ impl RaftEngineConfig {
         Ok(())
     }
 
+    fn optimize_for(&mut self, raft_store: &RaftstoreConfig, raft_kv_v2: bool) {
+        if !raft_kv_v2 {
+            let cur_batch_compression_thd = self.config().batch_compression_threshold;
+            let adaptive_batch_comp_thd = RaftEngineReadableSize(std::cmp::max(
+                cur_batch_compression_thd.0 / (raft_store.store_io_pool_size + 1) as u64,
+                RaftEngineReadableSize::kb(4).0,
+            ));
+            // As the async-io is enabled by default (raftstore.store_io_pool_size == 1),
+            // testing records shows that using 4kb as the default value can achieve
+            // better performance and reduce the IO overhead.
+            // Meanwhile, the batch_compression_threshold cannot be modified dynamically if
+            // the threads count of async-io are changed manually.
+            if !self.customized && raft_store.store_io_pool_size > 0 {
```

Comment:
you can make `batch_compression_threshold` as `Option<..>`, then you can know whether it's sepecified

Reply:
Not appropriate.

> make batch_compression_threshold as Option<..>, then you can know whether it's sepecified
If so, we cannot know whether the user forcely disable compression feature in RaftEngine, as the `None` is used as the default value for `batch_compression_threshold`. Moreover, RaftEngine also needs a default value with `Some(8kb)` and it is not compatible to this proposal.